src/rebar_prv_eunit.erl

```diff
@@ -134,17 +138,28 @@ resolve_tests(State) ->
     Files        = resolve(file, RawOpts),
     Modules      = resolve(module, RawOpts),
     Suites       = resolve(suite, module, RawOpts),
-    Apps ++ Applications ++ Dirs ++ Files ++ Modules ++ Suites.
+    Generator    = resolve(generator, RawOpts),
+    Apps ++ Applications ++ Dirs ++ Files ++ Modules ++ Suites ++ Generator.
 
 resolve(Flag, RawOpts) -> resolve(Flag, Flag, RawOpts).
 
 resolve(Flag, EUnitKey, RawOpts) ->
     case proplists:get_value(Flag, RawOpts) of
         undefined -> [];
-        Args      -> lists:map(fun(Arg) -> normalize(EUnitKey, Arg) end,
-                               rebar_string:lexemes(Args, [$,]))
+        Args      ->
+            lists:flatten(lists:map(fun(Arg) -> normalize(EUnitKey, Arg) end,
+                                    rebar_string:lexemes(Args, [$,])))
     end.
 
+normalize(generator, Value) ->
+    case string:tokens(Value, [$:]) of
+        [Module0, Functions] ->
+            Module = list_to_atom(Module0),
+            lists:map(fun(F) -> {generator, Module, list_to_atom(F)} end,
+                      string:tokens(Functions, [$;]));
```

Comment:
I would possibly use `normalize(...) -> ...` as a wrapper for a secondary `normalize_(...)` call to just make it an internal contract that a flat list is returned.

Reply:
Thanks for your assistant.

memo: we can write them like this.
```erlang
normalize(EUnitKey, Args) ->
    F = case EUnitKey of
            generator -> flatmap;
            _         -> map
        end,
    lists:F(fun(Arg) -> normalize_(EUnitKey, Arg) end, Args).
```
It is not easy for me. However, it can be used if you accept it.

Reply:
Current form is fine to me.